### PR TITLE
Wreck materials

### PR DESCRIPTION
**Pull request by slkendell-mt**
- Need to look over workflow again. 

- Heavy JS - Also not ideal 

- Need to update View to display all 
- Need to update form layout. 

@SamBrownMT - Have a look over the initial workflow. Would like to get your thoughts.

### CODE REVIEW

File: backoffice/src/Models/ViewModels/DroitView.cs [hunk 1]
```diff
@@ -27,6 +27,12 @@ public DroitView(Droit droit)
             Wreck = new WreckView(droit.Wreck);
         }
 
+        WreckMaterial = new WreckMaterialView();
+
+        if(droit.WreckMaterials.Any()){
+            WreckMaterial = new WreckMaterialView(droit.WreckMaterials.First());
```
Comment (slkendell-mt) [inline backoffice/src/Models/ViewModels/DroitView.cs:33]: Ignore this for now, still in phase show a single one.

File: backoffice/src/Services/WreckService.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public async Task<Wreck> SaveWreckAsync(Wreck wreck)
     public async Task<Wreck> GetWreckAsync(Guid id) =>
         await _repo.GetWreckAsync(id);
 
-    public async Task<Guid> GetWreckIdAsync(WreckForm wreckForm)
+    public async Task<Guid> SaveWreckFormAsync(WreckForm wreckForm)
```
Comment (SamBrownMT) [inline backoffice/src/Services/WreckService.cs:45]: I wonder if this should be separated out into two methods (for following single responsibility principle). Seems OTT for the size of the methods but could improve readability.

Comment (slkendell-mt) [reply backoffice/src/Services/WreckService.cs:45]: I generally like having a single function -> Upsert

That then works out whether it should Update or Insert (Add)

Then calls either Update or Add

Comment (slkendell-mt) [reply backoffice/src/Services/WreckService.cs:45]: Then Update/Add are private and never called directly externally

**Review by SamBrownMT — APPROVED** [review]

**Pull request merged**